src/components/Landing/EventCard/EventCard.jsx

```diff
@@ -1,73 +1,75 @@
-import React, { useState, useEffect, useMemo } from "react";
+import React, { useState, useEffect } from "react";
 import { NavLink } from "react-router-dom";
 import PropTypes from "prop-types";
 import { BiRightArrowAlt } from "react-icons/bi";
 import moment from "moment";
-import IconsEventCard from "./IconsEventCard/IconsEventCard";
 import "./EventCard.scss";
 import PhotoService from "../../../services/PhotoService";
 import {
   EVENT_DEFAULT_IMAGE,
   FORMATS,
 } from "../../../constants/eventConstants";
 
-const EventCard = ({ event }) => {
+const EventCard = ({ event, additionalButtons, additionalModal }) => {
   const { id, title, dateFrom } = event;
-  const photoService = useMemo(() => new PhotoService(), []);
+  const photoService = new PhotoService();
   const [eventImage, setEventImage] = useState(EVENT_DEFAULT_IMAGE);
-
+  const sliceLength = title.length > 35 ? `${title.slice(0, 35)}...` : title;
   useEffect(() => {
-    // TODO move to service and call from action
     photoService.getPreviewEventPhoto(id).then(eventPreviewImage => {
-      if (eventPreviewImage) {
-        const image = URL.createObjectURL(eventPreviewImage);
-        setEventImage(image);
+      if (eventPreviewImage !== null) {
+        setEventImage(URL.createObjectURL(eventPreviewImage));
       }
     });
```

Comment:
Move this logic from component

Reply:
This component refactoring by another teammate